sound/soc/sof/topology.c

```diff
@@ -1588,8 +1588,12 @@ static int sof_link_load(struct snd_soc_component *scomp, int index, struct snd_
 	if (!slink)
 		return -ENOMEM;
 
-	slink->hw_configs = cfg->hw_config;
 	slink->num_hw_configs = le32_to_cpu(cfg->num_hw_configs);
+	slink->hw_configs = kmemdup(cfg->hw_config,
+				    sizeof(*slink->hw_configs) * slink->num_hw_configs,
+				    GFP_KERNEL);
```

Comment:
Should we free slink->hw_configs in the `err:` label?

Reply:
yes, thanks for pointing out. fixed all the error paths now